components/shapediver/viewport/ViewportHistoryButtons.tsx

```diff
@@ -0,0 +1,272 @@
+
+interface Props {
+	/**
+	 * Style properties for the container
+	 */
+	style?: React.CSSProperties;
+	/**
+	 * Size of the action icons
+	 */
+	size?: number;
+	/**
+	 * Icon color
+	 */
+	color?: string;
+	/**
+	 * Icon color when disabled
+	 */
+	colorDisabled?: string;
+	/**
+	 * Button variant
+	 */
+	variant?: string;
+	/**
+	 * Button variant when disabled
+	 */
+	variantDisabled?: string;
+	/**
+	 * Style for individual icons
+	 */
+	iconStyle?: MantineStyleProp;
+	/**
+	 * Namespace of the session
+	 */
+	namespace: string;
+}
+
+const defaultProps: Required<Omit<Props, "namespace">> = {
+	style: {display: "flex", gap: "4px"},
+	size: 32,
+	color: "black",
+	colorDisabled: "grey",
+	variant: "subtle",
+	variantDisabled: "transparent",
+	iconStyle: {m: "3px"},
+};
+
+/**
+ * Component that provides undo/redo buttons and context menu for viewport history management.
+ * Includes browser history navigation and parameter/model state management.
+ * Buttons are disabled when there are pending parameter changes that need to be accepted or rejected.
+ */
+export default function ViewportHistoryButtons(props: Props) {
+	const {
+		style,
+		size,
+		color,
+		colorDisabled,
+		variant,
+		variantDisabled,
+		iconStyle,
+		namespace,
+	} = {...defaultProps, ...props};
+
+	const [isImportDialogOpen, setIsImportDialogOpen] = useState(false);
+
+	const {canGoBack, canGoForward, goBack, goForward, isLoading} =
+		useViewportHistory();
+
+	const notifications = useContext(NotificationContext);
+
+	const {exportParameters, importParameters, resetParameters} =
+		useParameterImportExport(namespace);
+
+	const {importModelState, isLoading: isModelStateLoading} =
+		useImportModelState(namespace);
+
+	const [isCreatingModelState, setIsCreatingModelState] = useState(false);
+	const {createModelState, applyModelStateToQueryParameter} =
+		useCreateModelState({namespace});
+
+	const onCreateModelState = useCallback(async () => {
+		setIsCreatingModelState(true);
+		const {modelStateId} = await createModelState(
+			undefined, // <-- parameterNamesToInclude: use default according to the theme
+			undefined, // <-- parameterNamesToExclude: use default according to the theme
+			true, // <-- includeImage,
+			undefined, // <-- custom data
+			false, // <-- includeGltf
+		);
+
+		if (modelStateId) {
+			applyModelStateToQueryParameter(modelStateId);
+			notifications.success({
+				message: "Model state successfully created",
+			});
+		}
+
+		setIsCreatingModelState(false);
+	}, []);
+
+	const executing = useShapeDiverStoreParameters((state) => {
+		const ids = state.sessionDependency[namespace];
+
+		return !ids.every((id) => !state.parameterChanges[id]?.executing);
+	});
+
+	const parameterChanges = useShapeDiverStoreParameters(
+		useCallback(
+			(state) =>
+				Object.keys(state.parameterChanges).reduce((acc, id) => {
+					acc.push(state.parameterChanges[id]);
+					return acc;
+				}, [] as IParameterChanges[]),
+			[],
+		),
+	);
```

Comment:
@LemToUp small optimization possibility: `executing` can probably be defined based on `parameterChanges`. 
Great idea to use `useCallback` for the state selector. I wasn't aware that this is possible.